src/figuresClasses.ts

```diff
@@ -1,11 +1,107 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    const triangleSmallerSides = [a, b, c]
+      .filter((num) => num !== Math.max(a, b, c))
+      .reduce((cur, next) => cur + next, 0);
+
+    this.shape = 'triangle';
+    this.color = color;
+
+    if (Math.max(a, b, c) >= triangleSmallerSides) {
+      throw new Error(
+        'Longer side cannot be longer or equal of summury of smallest sides',
+      );
+    } else if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side cannot be 0');
+    } else {
+      this.a = a;
+      this.b = b;
+      this.c = c;
```

Comment:
The constructor manually assigns values to class properties which is redundant when using TypeScript's parameter properties. You can simplify the constructor by using access modifiers (public or private) in the parameter list, and TypeScript will automatically create and assign those properties.